skfda/exploratory/stats/_functional_transformers.py

```diff
@@ -75,3 +75,112 @@ def local_averages(
         for i in range(n_intervals)
     ]
     return np.asarray(integrated_data)
+
+
+def number_up_crossings(
+    data: FDataGrid,
+    levels: np.ndarray,
+) -> np.ndarray:
+    r"""
+    Calculate the number of up crossings to a level of a FDataGrid.
+
+    Let f_1(X) = N_i, where N_i is the number of up crossings of X
+    to a level c_i \in \mathbb{R}, i = 1,\dots,p.
+
+    Recall that the process X(t) is said to have an up crossing of c
+    at :math:`t_0 > 0` if for some :math:`\epsilon >0`, X(t) $\leq$
+    c if t :math:'\in (t_0 - \epsilon, t_0) and X(t) $\geq$ c if
+    :math:`t\in (t_0, t_0+\epsilon)`.
+
+    If the trajectories are differentiable, then
+    :math:`N_i = card\{t \in[a,b]: X(t) = c_i, X' (t) > 0\}.`
+
+        Args:
+            data: FDataGrid where we want to calculate
+            the number of up crossings.
+            levels: sequence of numbers including the levels
+            we want to consider for the crossings.
+        Returns:
+            ndarray of shape (n_levels, n_samples)\
+            with the values of the counters.
+
+    Example:
+
+    We import the Medflies dataset and for simplicity we use
+    the first 50 samples.
+    >>> from skfda.datasets import fetch_medflies
+    >>> dataset = fetch_medflies()
+    >>> X = dataset['data'][:50]
+
+    Then we decide the level we want to consider (in our case 40)
+    and call the function with the dataset. The output will be the number of
+    times each curve cross the level 40 growing.
+    >>> from skfda.exploratory.stats import number_up_crossings
+    >>> import numpy as np
+    >>> number_up_crossings(X, np.asarray([40]))
+    array([[[6],
```

Comment:
Shouldn't the returned array have dimensions `n_samples` x `len(levels)`?

Reply:
I have prioritized len(levels) over n_samples Therefore shape will be (len(levels), n_samples, n_dimensions). Is it correct, or should I reshape it?

Reply:
In order for the result to be usable in sklearn it should be a 2d array and the first dimension should be `n_samples`.